saltproc/tests/test_simulation.py

```diff
@@ -1,41 +1,440 @@
+    """
+    This unit test checks that select entries that ``store_mat_data_()`
+    stores in the database match the corresponding entries from the input
+    explicity in value and implicitly in type.
+    """
+    # read data
+    mats_before = simulation.sim_depcode.read_dep_comp(
+        False)
+    mats_after = simulation.sim_depcode.read_dep_comp(
+        True)
+
+    fuel_before = mats_before['fuel']
+    pois_before = mats_before['ctrlPois']
+
+    fuel_after = mats_after['fuel']
+    pois_after = mats_after['ctrlPois']
+
+    # we want to keep the old path for other sims, but for this
+    # test we'll want a fresh db
+    db_path_old = simulation.db_path
+    db_file = serpent.iter_inputfile + '.h5'
+    simulation.db_path = db_file
+
+    # store data at end
+    simulation.store_mat_data(mats_before, 0, False)
+    simulation.store_mat_data(mats_after, 0, True)
+
+    # read stored data
+    try:
+        db = tb.open_file(simulation.db_path, mode='r',
+                          filters=simulation.compression_params)
+    except Exception:
+        print('Unable to assign correct value to db.\
+              See error stack for more info.')
+
+    tmats = db.root.materials
+
+    tfuel_before = tmats.fuel.before_reproc.comp[0]
+    tfuel_before_params = tmats.fuel.before_reproc.parameters[0]
+
+    tfuel_after = tmats.fuel.after_reproc.comp[0]
+    tfuel_after_params = tmats.fuel.after_reproc.parameters[0]
+
+    tpois_before = tmats.ctrlPois.before_reproc.comp[0]
+    tpois_before_params = tmats.ctrlPois.before_reproc.parameters[0]
 
+    tpois_after = tmats.ctrlPois.after_reproc.comp[0]
+    tpois_after_params = tmats.ctrlPois.after_reproc.parameters[0]
 
-def test_switch_to_next_geometry():
-    shutil.copy2(geo_test_input, serpent.input_path + '_test')
-    serpent.input_path = serpent.input_path + '_test'
-    simulation.switch_to_next_geometry()
-    d = serpent.read_depcode_template(serpent.input_path)
-    assert d[5].split('/')[-1] == '988.inp"\n'
-    os.remove(serpent.input_path)
+    # Check the mass fractions
+    try:
+        assert tfuel_before[1566] == fuel_before['U235']
+        assert tfuel_before[1570] == fuel_before['U238']
+        assert tfuel_before[37] == fuel_before['F19']
+        assert tfuel_before[8] == fuel_before['Li7']
+        assert tfuel_before[1610] == fuel_before['Cm240']
+        assert tfuel_before[1589] == fuel_before['Pu239']
+        assert tpois_before[1213] == pois_before['Gd155']
+        assert tpois_before[32] == pois_before['O16']
+
+        assert tfuel_after[1566] == fuel_after['U235']
+        assert tfuel_after[1570] == fuel_after['U238']
+        assert tfuel_after[37] == fuel_after['F19']
+        assert tfuel_after[8] == fuel_after['Li7']
+        assert tfuel_after[1610] == fuel_after['Cm240']
+        assert tfuel_after[1589] == fuel_after['Pu239']
+        assert tpois_after[1213] == pois_after['Gd155']
+        assert tpois_after[32] == pois_after['O16']
+
```

Comment:
Is there a way to make this indexing more flexible?  If the line-order of the file where the "tfuel_after" changes, won't it stop matching the test?

Reply:
To my knowledge, SaltProc doesn't actually store names/ids for nuclides in its database file, so you point about the indexing is completely valid. We absolutely should implement functionality for SaltProc to store the nuclide names so that we can query for information about specific nuclides.

Reply:
On second thought, I guess that's the entire point of the test. If the test doesn't pass than we know something is wrong. But I think we should still switch to a stronger method of retrieving information about nuclide composition.